### PR TITLE
GPC-367 Call enrich lambda

### PR DESCRIPTION
**Pull request by CarlyG55**

### CODE REVIEW

File: src/main/kotlin/uk/gov/gdx/datashare/services/GroApiService.kt [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    }
+
+    return parsedResponse.Event!!
+  }
+
+  private fun createEnrichEventPayload(dataId: String): String {
+    val jsonObj = JSONObject()
+    jsonObj["id"] = dataId
+    return jsonObj.toString()
+  }
```
Comment (CarlyG55) [inline src/main/kotlin/uk/gov/gdx/datashare/services/GroApiService.kt:33]: This was copied from aws's docs - couldn't really figure a nicer way to do it immediately with objectMapper but would be nice to keep it consistent

File: src/main/kotlin/uk/gov/gdx/datashare/services/GroApiService.kt [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      throw NoDataFoundException(dataId)
+    }
+
+    return parsedResponse.Event!!
+  }
+
+  private fun createEnrichEventPayload(dataId: String): String {
+    val jsonObj = JSONObject()
+    jsonObj["id"] = dataId
+    return jsonObj.toString()
```
Comment (CarlyG55) [inline src/main/kotlin/uk/gov/gdx/datashare/services/GroApiService.kt:32]: This was copied from aws's docs - couldn't really figure a nicer way to do it immediately with objectMapper but would be nice to keep it consistent

File: src/main/kotlin/uk/gov/gdx/datashare/services/GroApiService.kt [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    val jsonPayload = createEnrichEventPayload(dataId)
+
+    val res = lambdaService.invokeLambda("$environment-gro-ingestion-lambda-enrich-event", jsonPayload)
+    val parsedResponse = lambdaService.parseLambdaResponse(res, GroEnrichEventResponse::class.java)
+
+    if (parsedResponse.StatusCode.equals(HttpStatus.NOT_FOUND)) {
+      throw NoDataFoundException(dataId)
+    }
+
+    return parsedResponse.Event!!
```
Comment (CarlyG55) [inline src/main/kotlin/uk/gov/gdx/datashare/services/GroApiService.kt:26]: I don't like this assertion, but the check below takes out the null case

File: src/main/kotlin/uk/gov/gdx/datashare/services/GroApiService.kt [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+import net.minidev.json.JSONObject
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.http.HttpStatus
+import org.springframework.stereotype.Service
+import uk.gov.gdx.datashare.config.NoDataFoundException
+import uk.gov.gdx.datashare.models.GroDeathRecord
+import uk.gov.gdx.datashare.models.GroEnrichEventResponse
+
+@Service
+class GroApiService(
```
Comment (CarlyG55) [inline src/main/kotlin/uk/gov/gdx/datashare/services/GroApiService.kt:12]: This is probably a bad name

**Review by ethanmills — APPROVED** [review]

**Pull request merged**